content_scripts/content_annotation_handler.js

```diff
@@ -200,42 +370,53 @@ function createRegexMap(map) {
 	for (let key of Object.keys(map)) {
 		// We want a global, case-insensitive replacement.
 		// @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp
-		regexs.set(key, new RegExp(key, 'gi'));
+		try {
+			regexs.set(key, new RegExp(key, 'gi'));
+		} catch (err) {
+			// TODO: some regexs will be invalid
+			console.error(err);
+		}
 	}
 	return regexs;
 }
 
-browser.runtime.onMessage.addListener((message) => {
+function messageListener(message) {
 	if (message.intended_recipient !== "RECIPIENT_CONTENT") {
 		return; // don't send response so intended recipient can still pick up this message
 	}
 	if (message.command === "CONTENT_ANNOTATE") {
 		if (!message.MAPPINGS) {
-			return Promise.reject(Error("Bad request (missing MAPPINGS"));
+			return Promise.reject(Error("Bad request (missing `MAPPINGS`)"));
 		}
 		// TODO: allow for partial annotation
-		browser.runtime.sendMessage({
-			intended_recipient: "RECIPIENT_BACKGROUND",
-			command: "BACKGROUND_GET_CONSOLIDATED_MAPPING",
-			MAPPINGS_TO_CONSOLIDATE: message.MAPPINGS
-		})
-		.then((BACKGROUND_GET_CONSOLIDATED_MAPPING_response) => {
-			let regexs = createRegexMap(BACKGROUND_GET_CONSOLIDATED_MAPPING_response);
-			rubyWrap(document.body, regexs, BACKGROUND_GET_CONSOLIDATED_MAPPING_response);
-			return Promise.resolve();
+		return new Promise((resolve, reject) => {
+			smeagol.sendMessage({
+				intended_recipient: "RECIPIENT_BACKGROUND",
+				command: "BACKGROUND_GET_CONSOLIDATED_MAPPING",
+				MAPPINGS_TO_CONSOLIDATE: message.MAPPINGS
+			})
+			.then((BACKGROUND_GET_CONSOLIDATED_MAPPING_response) => {
+				let regexs = createRegexMap(BACKGROUND_GET_CONSOLIDATED_MAPPING_response);
+				rubyWrap(document.body, regexs, BACKGROUND_GET_CONSOLIDATED_MAPPING_response);
+				return resolve({});

```

Comment:
You're resolving with `{}` here and with nothing on line 411. Pick one and use it consistently. I'd prefer {}